crates/service/tests/router_test.rs

```diff
@@ -0,0 +1,128 @@
+        .tcp_nodelay(true)
+        .build()
+        .expect("Failed to init HTTP client");
+
+    let allocation = INDEXER_ALLOCATIONS
+        .values()
+        .collect::<Vec<_>>()
+        .pop()
+        .unwrap()
+        .clone();
+    let deployment = allocation.subgraph_deployment.id;
+
+    let mock_server = MockServer::start().await;
+
+    let mock = Mock::given(method("POST"))
+        .and(path(format!("/subgraphs/id/{deployment}")))
+        .respond_with(ResponseTemplate::new(200).set_body_raw(
+            r#"
+                {
+                    "data": {
+                        "graphNetwork": {
+                            "currentEpoch": 960
+                        }
+                    }
+                }
+                "#,
+            "application/json",
+        ));
+    mock_server.register(mock).await;
+
+    let (_escrow_tx, escrow_accounts) = watch::channel(EscrowAccounts::new(
+        test_assets::ESCROW_ACCOUNTS_BALANCES.clone(),
+        test_assets::ESCROW_ACCOUNTS_SENDERS_TO_SIGNERS.clone(),
+    ));
+    let (_dispute_tx, dispute_manager) = watch::channel(Address::ZERO);
+
+    let (_allocations_tx, allocations) = watch::channel(test_assets::INDEXER_ALLOCATIONS.clone());
+
+    let graph_node_url = Url::parse(&mock_server.uri()).unwrap();
+
+    let router = ServiceRouter::builder()
+        .database(database)
+        .domain_separator(TAP_EIP712_DOMAIN.clone())
+        .http_client(http_client)
+        .graph_node(leak(GraphNodeConfig {
+            query_url: graph_node_url.clone(),
+            status_url: graph_node_url.clone(),
+        }))
+        .indexer(leak(IndexerConfig {
+            indexer_address: *test_assets::INDEXER_ADDRESS,
+            operator_mnemonic: test_assets::INDEXER_MNEMONIC.clone(),
+        }))
+        .service(leak(indexer_config::ServiceConfig {
+            serve_network_subgraph: false,
+            serve_escrow_subgraph: false,
+            serve_auth_token: None,
+            host_and_port: "0.0.0.0:0".parse().unwrap(),
+            url_prefix: "/".into(),
+            tap: indexer_config::ServiceTapConfig {
+                max_receipt_value_grt: NonZeroGRT::new(1000000000000).unwrap(),
+            },
+            free_query_auth_token: None,
+        }))
+        .blockchain(leak(BlockchainConfig {
+            chain_id: indexer_config::TheGraphChainId::Test,
+            receipts_verifier_address: *test_assets::VERIFIER_ADDRESS,
+        }))
+        .timestamp_buffer_secs(Duration::from_secs(10))
+        .escrow_accounts(escrow_accounts)
+        .dispute_manager(dispute_manager)
+        .allocations(allocations)
+        .build();
+
+    let app = router.create_router().await.unwrap();
+
+    let timestamp = SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .expect("Time went backwards")
+        .as_nanos();
+
+    let receipt = create_signed_receipt(allocation.id, 0, timestamp as u64, 100).await;
```

Comment:
Please name the arguments. I have no idea what the inputs are without reading the function signature.